Tests/test_pyarrow.py

```diff
@@ -110,3 +149,59 @@ def test_lifetime2() -> None:
     px = img2.load()
     assert px  # make mypy happy
     assert isinstance(px[0, 0], int)
+
+
+UINT_ARR = (fl_uint8_4_type, [1, 2, 3, 4], 1)
+UINT = (pyarrow.uint8(), 3, 4)
+INT32 = (pyarrow.uint32(), 0xABCDEF45, 1)
+
+
+@pytest.mark.parametrize(
+    "mode, data_tp, mask",
+    (
+        ("L", (pyarrow.uint8(), 3, 1), None),
+        ("I", (pyarrow.int32(), 1 << 24, 1), None),
+        ("F", (pyarrow.float32(), 3.14159, 1), None),
+        ("LA", UINT_ARR, [0, 3]),
+        ("LA", UINT, [0, 3]),
+        ("RGB", UINT_ARR, [0, 1, 2]),
+        ("RGBA", UINT_ARR, None),
+        ("CMYK", UINT_ARR, None),
+        ("YCbCr", UINT_ARR, [0, 1, 2]),
+        ("HSV", UINT_ARR, [0, 1, 2]),
+        ("RGB", UINT, [0, 1, 2]),
+        ("RGBA", UINT, None),
+        ("CMYK", UINT, None),
+        ("YCbCr", UINT, [0, 1, 2]),
+        ("HSV", UINT, [0, 1, 2]),
+    ),
+)
+def test_fromarray(mode: str, data_tp: tuple, mask: list[int] | None) -> None:
```

Comment:
```suggestion
def test_fromarray(
    mode: str, data_tp: tuple[Any, list[int] | float, int], mask: list[int] | None
) -> None:
```

Reply:
To channel Raymond Hettinger -- Why not a named tuple?